fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPrepare.java

```diff
@@ -0,0 +1,1917 @@
+import com.google.common.collect.Sets;
+import com.starrocks.catalog.ResourceGroup;
+import com.starrocks.catalog.ResourceGroupClassifier;
+import com.starrocks.common.Config;
+import com.starrocks.common.Reference;
+import com.starrocks.common.UserException;
+import com.starrocks.common.util.DebugUtil;
+import com.starrocks.common.util.ListUtil;
+import com.starrocks.planner.DataPartition;
+import com.starrocks.planner.DataSink;
+import com.starrocks.planner.DataStreamSink;
+import com.starrocks.planner.DeltaLakeScanNode;
+import com.starrocks.planner.ExchangeNode;
+import com.starrocks.planner.ExportSink;
+import com.starrocks.planner.FileTableScanNode;
+import com.starrocks.planner.HdfsScanNode;
+import com.starrocks.planner.HudiScanNode;
+import com.starrocks.planner.IcebergScanNode;
+import com.starrocks.planner.JoinNode;
+import com.starrocks.planner.MultiCastDataSink;
+import com.starrocks.planner.MultiCastPlanFragment;
+import com.starrocks.planner.OlapScanNode;
+import com.starrocks.planner.PlanFragment;
+import com.starrocks.planner.PlanFragmentId;
+import com.starrocks.planner.PlanNode;
+import com.starrocks.planner.PlanNodeId;
+import com.starrocks.planner.ResultSink;
+import com.starrocks.planner.RuntimeFilterDescription;
+import com.starrocks.planner.ScanNode;
+import com.starrocks.planner.UnionNode;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.service.FrontendOptions;
+import com.starrocks.sql.common.ErrorType;
+import com.starrocks.sql.common.StarRocksPlannerException;
+import com.starrocks.system.Backend;
+import com.starrocks.system.ComputeNode;
+import com.starrocks.thrift.InternalServiceVersion;
+import com.starrocks.thrift.TDescriptorTable;
+import com.starrocks.thrift.TEsScanRange;
+import com.starrocks.thrift.TExecBatchPlanFragmentsParams;
+import com.starrocks.thrift.TExecPlanFragmentParams;
+import com.starrocks.thrift.THdfsScanRange;
+import com.starrocks.thrift.TInternalScanRange;
+import com.starrocks.thrift.TLoadJobType;
+import com.starrocks.thrift.TNetworkAddress;
+import com.starrocks.thrift.TPlanFragmentDestination;
+import com.starrocks.thrift.TPlanFragmentExecParams;
+import com.starrocks.thrift.TQueryGlobals;
+import com.starrocks.thrift.TQueryOptions;
+import com.starrocks.thrift.TQueryType;
+import com.starrocks.thrift.TRuntimeFilterParams;
+import com.starrocks.thrift.TScanRangeLocation;
+import com.starrocks.thrift.TScanRangeLocations;
+import com.starrocks.thrift.TScanRangeParams;
+import com.starrocks.thrift.TUniqueId;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+public class CoordinatorPrepare {
```

Comment:
CoordinatorPrepare is not a good name. Prepare is a verb. use Preprocessor instead?

Reply:
done